### PR TITLE
Specify post type when syncing a wpcom blog.

### LINKED ISSUE #3428: Post/Page revisions appear for wpcom blogs
A change in the v1.1 REST API returns any white listed post type, including revisions.  

In the app when a blogs content is synced from the blog details screen it queries for new posts and specifies a type of "any".  The idea was to fetch posts and pages in one pass, however we don't want the revisions.

### PR DESCRIPTION
**Pull request by aerych**
Avoids fetching revisions and other unwanted post types.

Closes #3428

### CODE REVIEW

**Comment by aerych** [comment]
cc @bummytime for review

File: WordPress/Classes/Services/BlogService.m [hunk 1]
```diff
@@ -274,10 +274,13 @@ - (void)syncBlog:(Blog *)blog
                                    failure:nil];
 
     PostService *postService = [[PostService alloc] initWithManagedObjectContext:self.managedObjectContext];
-    // FIXME: this is hacky, but XML-RPC doesn't support fetching "any" type of post
-    // Ideally we'd do a multicall and fetch both posts/pages, but it's out of scope for this commit
+    // FIXME: this is hacky, ideally we'd do a multicall and fetch both posts/pages, but it's out of scope for this commit
     if (blog.restApi) {
```
Comment (bummytime) [inline WordPress/Classes/Services/BlogService.m]: It looks like the else clause is identical to this. Do we need this anymore?

Comment (aerych) [inline WordPress/Classes/Services/BlogService.m]: Why, no.  No we do not!

**Comment by bummytime** [comment]
@aerych tested on iOS 7 and 8. The fix alleviates the crazy 5+ post duplication we were seeing. I still see our "old" duplicate post issue:

screen shot 2015-03-18 at 10 41 36 am

scaled screen shot 2015-03-18 at 10 38 25 am

But it is out of scope for this fix. Nice work on sleuthing this. Lemme know about my comment above.

**Comment by aerych** [comment]
> I still see our "old" duplicate post issue

Yeah. Seems we have a few different flavors of that bug.  We'll find and squash 'em all eventually

**Comment by bummytime** [comment]
:shipit:

**Comment by aerych** [comment]
Thanks @bummytime

**Pull request merged**